mist-common/src/main/java/edu/snu/mist/common/operators/ApplyStatefulOperator.java

```diff
@@ -86,16 +95,37 @@ public void processLeftWatermark(final MistWatermarkEvent input) {
     outputEmitter.emitWatermark(input);
     if (input.isCheckpoint()) {
       latestCheckpointTimestamp = input.getTimestamp();
+      final Map<String, Object> stateMap = new HashMap<>();
+      stateMap.put("applyStatefulFunctionState", applyStatefulFunction.getCurrentState());
+      checkpointMap.put(input.getTimestamp(), stateMap);
     }
   }
 
   @Override
-  public Map<String, Object> getOperatorState() {
+  public Map<String, Object> getCurrentOperatorState() {
```

Comment:
Why don't you just return `stateMap`?

Reply:
`stateMap` is not a class variable that always exists.

Reply:
How about getStateSnapshot()?